libpod/networking_linux.go

```diff
@@ -981,3 +983,139 @@ func (w *logrusDebugWriter) Write(p []byte) (int, error) {
 	logrus.Debugf("%s%s", w.prefix, string(p))
 	return len(p), nil
 }
+
+// DisconnectContainerFromNetwork removes a container from its CNI network
+func (r *Runtime) DisconnectContainerFromNetwork(nameOrID, netName string, force bool) error {
+	ctr, err := r.LookupContainer(nameOrID)
+	if err != nil {
+		return err
+	}
+
+	networks, err := ctr.networksByNameIndex()
+	if err != nil {
+		return err
+	}
+
+	exists, err := network.Exists(r.config, netName)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return errors.Wrap(define.ErrNoSuchNetwork, netName)
+	}
+
+	index, nameExists := networks[netName]
+	if !nameExists && len(networks) > 0 {
+		return errors.Errorf("container %s is not connected to network %s", nameOrID, netName)
+	}
+
+	ctr.lock.Lock()
+	defer ctr.lock.Unlock()
+	if err := ctr.syncContainer(); err != nil {
+		return err
+	}
+
+	podConfig := r.getPodNetwork(ctr.ID(), ctr.Name(), ctr.state.NetNS.Path(), []string{netName}, ctr.config.PortMappings, nil, nil)
+	if err := r.netPlugin.TearDownPod(podConfig); err != nil {
+		return err
+	}
+	if err := r.state.NetworkDisconnect(ctr, netName); err != nil {
+		return err
+	}
+
+	// update network status
+	networkStatus := ctr.state.NetworkStatus
+	// if len is one and we confirmed earlier that the container is in
+	// fact connected to the network, then just return an empty slice
+	if len(networkStatus) == 1 {
+		ctr.state.NetworkStatus = make([]*cnitypes.Result, 0)
+	} else {
+		// clip out the index of the network
+		networkStatus[len(networkStatus)-1], networkStatus[index] = networkStatus[index], networkStatus[len(networkStatus)-1]
+		// shorten the slice by one
+		ctr.state.NetworkStatus = networkStatus[:len(networkStatus)-1]
+	}
+	return nil
+}
+
+// ConnectContainerToNetwork connects a container to a CNI network
+func (r *Runtime) ConnectContainerToNetwork(nameOrID, netName string, aliases []string) error {
+	ctr, err := r.LookupContainer(nameOrID)
+	if err != nil {
+		return err
+	}
+
+	networks, err := ctr.networksByNameIndex()
```

Comment:
Same as above, splitting this bit on into a per-container function? I'd like to be able to call this on an existing container

Reply:
@mheon has agreed to allow me to defer this until my next PR